sw/host/hsmtool/src/util/attribute/attr.rs

```diff
@@ -321,21 +344,29 @@ impl AttributeMap {
         let mut map = AttributeMap::from(attrs.as_slice());
 
         let sensitive = map.get(&AttributeType::Sensitive);
-        if sensitive.is_none() || matches!(sensitive, Some(AttrData::Bool(false))) {
-            // If the object is not sensitive, we can read the value attribute.
-            let value = session
-                .get_attributes(object, &[cryptoki::object::AttributeType::Value])?
-                .remove(0);
-            let (ty, val) = into_kv(&value)?;
-            map.insert(ty, val);
-        } else {
-            // If the object is sensitive, we redact the value attribute.
-            map.insert(
-                AttributeType::Value,
-                AttrData::Redacted(Redacted::RedactedByHsm),
-            );
-        }
+        let object_sensitive =
+            sensitive.is_some() && matches!(sensitive, Some(AttrData::Bool(true)));
```

Comment:
Checking `is_some` should not be necessary here.  If sensitive is `None`, it will fail the pattern match against `Some(AttrData::Bool(true))`.

Reply:
Fixed.